src/commands/map.ts

```diff
@@ -0,0 +1,195 @@
+import type { ProfileDocumentNode, ProviderJson } from '@superfaceai/ast';
+import { parseDocumentId, parseProfile, Source } from '@superfaceai/parser';
+import Listr from 'listr';
+import { basename } from 'path';
+
+import type { Flags } from '../common/command.abstract';
+import { Command } from '../common/command.abstract';
+import type { UserError } from '../common/error';
+import { buildMapPath, buildProfilePath } from '../common/file-structure';
+import { exists, readFile } from '../common/io';
+import type { ILogger } from '../common/log';
+import { OutputStream } from '../common/output-stream';
+import { mapProviderToProfile } from '../logic/map';
+import { resolveProviderJson } from './new';
+
+export default class Map extends Command {
+  // TODO: add description
+  public static description =
+    'This commands uses Conlink profile and provider definition from `superface` folder and generate JS map and boilerplate code. Created integration is saved in `superface` folder and is ready to be used by our WASM SDK. User should check security, integration parameters and input before execution. Created integration can be tested by running `execute` command';
+
+  public static examples = [
+    'superface map <provider-name> <optional-profile-scope>.<profile-name>.profile',
+  ];
+
+  public static args = [
+    {
+      name: 'providerName',
+      description: 'Name of provider.',
+      required: true,
+    },
+    {
+      name: 'profileId',
+      description: 'Id of profile, eg: starwars.character-information',
+      required: false,
+    },
+  ];
+
+  public static flags = {
+    ...Command.flags,
+  };
+
+  public async run(): Promise<void> {
+    const { flags, args } = this.parse(Map);
+    await super.initialize(flags);
+    await this.execute({
+      logger: this.logger,
+      userError: this.userError,
+      flags,
+      args,
+    });
+  }
+
+  public async execute({
+    logger,
+    userError,
+    flags,
+    args,
+  }: {
+    logger: ILogger;
+    userError: UserError;
+    flags: Flags<typeof Map.flags>;
+    args: { providerName?: string; profileId?: string };
+  }): Promise<void> {
+    const { providerName, profileId } = args;
+
+    let map: string;
+    let providerJson: ProviderJson;
+    let profile: { source: string; name: string; scope: string | undefined };
+
+    const tasks = new Listr<{
+      providerName: string | undefined;
+      profileId: string | undefined;
+      quiet: boolean | undefined;
+    }>([
+      {
+        title: 'Loading Comlink interface',
```

Comment:
suggestion: I would use `profile` term here, as the command argument `profileId`
```suggestion
        title: 'Loading profile',
```